ai-service/src/ai_service/embeddings/encoding.py

```
@@ -0,0 +1,105 @@
+                texts,
+                convert_to_numpy=True,
+                normalize_embeddings=True,
+                batch_size=batch_size,
+                precision="float32",
+                show_progress_bar=show_progress_bar,
+                device=None,  # Use model's configured device
+            )
+        return cast(list[list[float]], embeddings.tolist())  # type: ignore
+
+    except Exception as e:
+        raise errors.EmbeddingError(f"Failed to encode texts: {e}") from e
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)